From: Tsutomu Itoh <t-itoh@jp.fujitsu.com>
Date: Mon, 1 Oct 2012 03:07:15 -0600
Patch-mainline: 3.7
Git-commit: 7a2d6a64645b38d7040bbd031c7a7b2655f5d976
References: FATE#312888
Subject: [PATCH] Btrfs: remove unnecessary IS_ERR in
 bio_readpage_error()

Because the value of extent_map is only a correct value or NULL,
so IS_ERR is unnecessary.

Signed-off-by: Tsutomu Itoh <t-itoh@jp.fujitsu.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/extent_io.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/btrfs/extent_io.c
+++ b/fs/btrfs/extent_io.c
@@ -2111,7 +2111,7 @@ static int bio_readpage_error(struct bio
 		}
 		read_unlock(&em_tree->lock);
 
-		if (!em || IS_ERR(em)) {
+		if (!em) {
 			kfree(failrec);
 			return -EIO;
 		}
